docs/dev/workflow-code-paths.rst

```diff
@@ -0,0 +1,63 @@
+Workflow Code Paths
+###################
+
+The intention of the following section is to give an overview of important functions and code paths executed to perform certain actions.
+
+NEURON simulations
+------------------
+
+A summary of the workflow of a typical `NEURON` simulation can be given with the following graph:
```

Comment:
Minor style nitpick: if you want `preformatted/code-like` text in rST you need to use double backticks, single backticks give _italic_ text.